### ISSUE TITLE
Support VLAN trunking in network configuration

### ISSUE THREAD

**Issue by celskeggs** [sipb/homeworld#340]
The new server environment involves network ports that use VLAN trunking, which means we need to reconfigure how we set up server networks.

Here's what a manually patched /etc/network/interfaces file looks like:

```
# This file describes the network interfaces available on your system
# and how to activate them. For more information, see interfaces(5).

source /etc/network/interfaces.d/*

# The loopback network interface
auto lo
iface lo inet loopback

# The primary network interface
allow-hotplug eno1
iface eno1 inet manual

auto eno1.240
iface eno1.240 inet static
	address 18.85.240.10/20
	gateway 18.85.240.1
	# dns-* options are implemented by the resolvconf package, if installed
	dns-nameservers 18.70.0.160 18.71.0.151 18.72.0.3
	dns-search unassigned-domain

auto eno1.241
iface eno1.241 inet static
	address 192.168.240.110/24
```

VLAN 240 is the publicly routable VLAN. VLAN 241 is the IPMI VLAN, so that block is only included on the supervisor node.

We should add appropriate configuration options for VLAN trunking in the cluster configuration file -- we want to both be able to control whether VLAN trunking is used and which VLAN numbers are used.

Eventually, we should make sure that servers can have this configured heterogenously.

**Comment by celskeggs** [comment]
Next step on this is to figure out how to get the Debian installer to use VLANs.

**Comment by celskeggs** [comment]
Twelve years after VLAN support in d-i was first requested, there's still no support. https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=433568

**Comment by celskeggs** [comment]
Fixed via #440 and tested on real hardware.
